packages/server/src/synchronizer/index.ts

```diff
@@ -53,6 +53,7 @@ export async function synchronizeFiles({
     }
 
     const readyHandler = () => {
+      reporter.stream.write({type: IDLE, payload: null})
       reporter.stream.write({type: READY, payload: null})
```

Comment:
Only thing I would be thinking maybe we might do well to send these events from within the actual stream? What do you think?

Reply:
This handler is mainly to help with promise conversion

Reply:
Also if we do that we don't actually need to pass a callback into the stream at all we can simply attach an `on('data', ...)` listener to the reporter stream.

Reply:
I think the above is fine for now. 

In the future not in this PR we could make an 'EventStream' decorator that accepts a stream and does the switching for us.

```ts
reporter.on(READY, (payload) => {
  // ...
})
```

Brainstorm 1: another option would be that we could use the fact that the stream is an eventemitter and simply define and emit events on the stream but I think that might have implications for our type system...

Reply:
@ryardley Would you mind finishing this? Struggling to understand the whole data flow of the server and want to focus on finishing the tutorial.

Reply:
I can certainly do it but I would love to understand what you need help with and see if there is some way I can create enough docs or resources to make it easier to grok.

Reply:
Finished the tutorial :). So, just not sure how to make READY fire once? Have played around with the order of firing events but I’m sure there’s something I’m missing,